File: sources/api/ActivityTrackerApi/Shared.Models/Import/HealthConnect/HealthConnectConfiguration.cs
using Shared.Enums;

namespace Shared.Models.Import.HealthConnect
{
    public class HealthConnectConfiguration
    {
        public Guid ConnectionGuid { get; set; }
        public ConnectionStatus Status { get; set; }
        public bool IsActive { get; set; }
        public bool IsInitialLoad { get; set; }
        public int SelectedSourceId { get; set; } = -1;
        public List<HealthConnectSource> AvailableSources { get; set; } = new List<HealthConnectSource>();
        public List<int> SelectedMetricIds { get; set; } = new List<int>();
        public List<HealthConnectMetric> Metrics { get; set; } = new List<HealthConnectMetric>();
        public string UpdatedAt { get; set; } = string.Empty;
        public string UpdatedBy { get; set; } = string.Empty;
    }
}

File: sources/api/ActivityTrackerApi/Core.WebApp/web-app/src/pages/healthConnect/types/HealthConnectConfiguration.ts
import { ConnectionStatus } from "../../../lib/enums/ConnectionStatus";
import { HealthConnectMetricMapping } from "./HealthConnectMetricMapping";
import { HealthConnectSourceMapping } from "./HealthConnectSourceMapping";

export type HealthConnectConfiguration = {
  id: number;
  deviceId: string;
  userId: number;
  deviceName: string;
  syncClientId: string;
  isActive: boolean;
  status: ConnectionStatus;
  sourceMappings: HealthConnectSourceMapping[];
  metricMappings: HealthConnectMetricMapping[];
  updatedAt: string;
  updatedBy: string;
};
